File: sensor_controller_py/sensor_controller_py/sensor_controller_server.py
from std_msgs.msg import Int64
from example_interfaces.srv import SetBool

import rclpy
from rclpy.node import Node

class SensorControllerServer(Node):
    def __init__(self):
        super().__init__("sensor_controller_server")

        self.service_ = self.create_service(SetBool, "command_service", self.server_cb)

    def server_cb(self, request, response):
        self.get_logger().info(f"Actuating signal is {request.data}")

        response.success = True
        return response

def main(args=None):
    rclpy.init(args=args)

    sensor_controller = SensorControllerServer()

    rclpy.spin(sensor_controller)

    sensor_controller.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: sensor_controller_cpp/src/sensor_controller_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using namespace std::placeholders;

class SensorControllerServer : public rclcpp::Node
{
  public:
    SensorControllerServer() : Node("sensor_controller_server")
    {
      service_ = this->create_service<example_interfaces::srv::SetBool>(
        "command_service", std::bind(&SensorControllerServer::server_cb, this, _1, _2));

    }

  private:
    void server_cb(const std::shared_ptr<example_interfaces::srv::SetBool::Request> request, std::shared_ptr<example_interfaces::srv::SetBool::Response> response)
    {
      RCLCPP_INFO(this->get_logger(), "Actuating signal is %d", request->data);

      response->success = true;
    }

    rclcpp::Service<example_interfaces::srv::SetBool>::SharedPtr service_;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  rclcpp::spin(std::make_shared<SensorControllerServer>());

  rclcpp::shutdown();
  return 0;
}